ext/puma_http11/extconf.rb

```diff
@@ -10,10 +10,7 @@
 
 unless ENV["PUMA_DISABLE_SSL"]
   # don't use pkg_config('openssl') if '--with-openssl-dir' is used
-  # also looks within the Ruby build for directory info
-  has_openssl_dir = dir_config('openssl').any? ||
-    RbConfig::CONFIG['configure_args']&.include?('openssl') ||
```

Comment:
My understanding is this line is not needed because `dir_config` should use `configure_args` already (although the code of `dir_config` and mkmf in general is not easy to read).
Also mentioned in https://github.com/ruby/openssl/issues/650#issuecomment-1645680989